prboom2/src/p_pspr.c

```diff
@@ -221,6 +221,73 @@ int weapon_preferences[2][NUMWEAPONS+1] = {
 // Center Weapon when Firing.
 int weapon_attack_alignment=0;
 
+// [XA] fixed version of P_SwitchWeapon that correctly
+// takes each weapon's ammotype and ammopershot into account,
+// instead of blindly assuming both.
+
+static int P_SwitchWeaponMBF21(player_t *player)
+{
+  int *prefer = weapon_preferences[0];
+  int currentweapon = player->readyweapon;
+  int newweapon = currentweapon;
+  int i = NUMWEAPONS+1;
+
+  weapontype_t checkweapon;
```

Comment:
C declarations again:
```c
int *prefer;
int currentweapon, newweapon;
int i;
weapontype_t checkweapon;
ammotype_t ammotype;

prefer = weapon_preferences[0];
currentweapon = player->readyweapon;
newweapon = currentweapon;
i = NUMWEAPONS + 1;
```

If you want to fix that in the original method too, I'm all for it, but at least let's have it fixed in the new code.

Reply:
Fixed both, since I was in the neighborhood. :P